AEPCampaignClassic/Sources/CampaignClassic+PublicAPI.swift

```diff
@@ -16,4 +16,89 @@ import Foundation
 
 /// Defines the public interface for the CampaignClassic extension
 @objc public extension CampaignClassic {
+
+    /// Registers a device with the configured Adobe Campaign Classic server instance.
+    /// The `callback` will be executed when the registration request is completed, returning success status of the call.
+    ///
+    /// - Parameters:
+    ///    - token : A unique device token received after registering your app with APNs servers
+    ///    - userKey : A `string` containing the user identifier
+    ///    - additionalParameters : a dictionary of custom key-value pairs to be sent along with the registration call
+    ///    - callback : a block which will be called after the device registration is complete. The callback returns
+    ///                YES if the registration request completed successfully, NO otherwise
+    static func registerDevice(token: Data, userKey: String, additionalParameter: [String: String]?, callback: @escaping (Bool) -> Void) {
+        let deviceInfo = [CampaignClassicConstants.EventDataKeys.CampaignClassic.DEVICE_INFO_KEY_DEVICE_NAME: UIDevice.current.name,
+                          CampaignClassicConstants.EventDataKeys.CampaignClassic.DEVICE_INFO_KEY_DEVICE_MODEL: UIDevice.current.model,
+                          CampaignClassicConstants.EventDataKeys.CampaignClassic.DEVICE_INFO_KEY_OS_NAME: UIDevice.current.systemName]
+
+        var eventData = [CampaignClassicConstants.EventDataKeys.CampaignClassic.REGISTER_DEVICE: true,
+                         CampaignClassicConstants.EventDataKeys.CampaignClassic.DEVICE_TOKEN: token.hexDescription,
+                         CampaignClassicConstants.EventDataKeys.CampaignClassic.USER_KEY: userKey,
+                         CampaignClassicConstants.EventDataKeys.CampaignClassic.DEVICE_INFO: deviceInfo] as [String: Any]
+
+        // attach additional parameters only if they are available
+        if let additionalParameter = additionalParameter {
+            eventData[CampaignClassicConstants.EventDataKeys.CampaignClassic.ADDITIONAL_PARAMETERS] = additionalParameter
+        }
+
+        let event = Event(name: "CampaignClassic Register Device",
+                          type: CampaignClassicConstants.SDKEventType.CAMPAIGN_CLASSIC,
+                          source: EventSource.requestContent,
+                          data: eventData)
+
+        MobileCore.dispatch(event: event, responseCallback: { responseEvent in
+            guard let responseEvent = responseEvent else {
+                callback(false)
```

Comment:
per the previous comment, it would be good to be able to indicate why registration has failed in this case